tensorflow/compiler/tf2tensorrt/kernels/trt_engine_op.cc

```diff
@@ -1057,9 +1068,11 @@ Status TRTEngineOp::GetEngineCacheResource(OpKernelContext* ctx,
     resource_name.remove_prefix(last_slash + 1);
   }
 
+  std::string resource(this->engine_name_prefix_);
+  resource += resource_name;
   // Get engine cache.
   return ctx->resource_manager()->LookupOrCreate(
-      std::string(kTfTrtContainerName), std::string(resource_name), cache_res,
+      std::string(kTfTrtContainerName), resource, cache_res,
```

Comment:
Regarding question (1) please use the engine_name_prefix_  as container name, and keep the resource name the same as before (the op name).
```suggestion
this->engine_name_prefix_, std::string(resource_name), cache_res, 
```

Reply:
@tfeher: DONE. I have implemented using container id. At the moment it is generated as a random string, but I will work on your other suggestion.